lib/std/os.zig

```diff
@@ -3199,8 +3199,16 @@ pub fn isatty(handle: fd_t) bool {
         if (isCygwinPty(handle))
             return true;
 
-        var out: windows.DWORD = undefined;
-        return windows.kernel32.GetConsoleMode(handle, &out) != 0;
+        // Support for ANSI escapes sequences was added to Windows 10 1511
+        // on 2016.
+        // See https://learn.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences.
+        const ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x0004;
+
+        var mode: windows.DWORD = undefined;
+        if (windows.kernel32.GetConsoleMode(handle, &mode) == 0) return false;
+
+        mode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;
+        return windows.kernel32.SetConsoleMode(handle, mode) != 0;
```

Comment:
1. Does this affect the use case of using `dumb` as argument to the windows console?
2. Did you compare how other applications are doing it, ie [Windows Terminal](https://github.com/microsoft/terminal)? It sounds very fishy, that 2 kernel32 calls are necessary to query the console state.

Reply:
> 1. Does this affect the use case of using `dumb` as argument to the windows console?
> 

I'm not aware of any `dumb` arguments for the windows console.  Note that I use Windows only for testing software compatibility.

>     2. Did you compare how other applications are doing it, ie [Windows Terminal](https://github.com/microsoft/terminal)? It sounds very fishy, that 2 kernel32 calls are necessary to query the console state.

AFAIK, Windows Terminal enables this the feature by default.

The problem, that I did not consider, is how to detect Windows Terminal.
With the current PR, escape codes will be disabled when a Zig program runs on a Windows Terminal, since calling `GetConsoleMode` will fail (**not tested**).

See https://github.com/microsoft/terminal/issues/1040.

Thanks.

Reply:
> 1. Does this affect the use case of using `dumb` as argument to the windows console?
> 
>     2. Did you compare how other applications are doing it, ie [Windows Terminal](https://github.com/microsoft/terminal)? It sounds very fishy, that 2 kernel32 calls are necessary to query the console state.

Note that the first call is to get the current state and the second it to set it.  It is the same as with Unix termios.

Also, about the Console API and Windows Terminal: calling `GetConsoleMode` should not fail.